src/PersonObjects/Person.ts

```diff
@@ -40,17 +42,184 @@ export abstract class Person implements IPerson {
   /** City that the person is in */
   city: CityName = CityName.Sector12;
 
-  gainHackingExp = personMethods.gainHackingExp;
-  gainStrengthExp = personMethods.gainStrengthExp;
-  gainDefenseExp = personMethods.gainDefenseExp;
-  gainDexterityExp = personMethods.gainDexterityExp;
-  gainAgilityExp = personMethods.gainAgilityExp;
-  gainCharismaExp = personMethods.gainCharismaExp;
-  gainIntelligenceExp = personMethods.gainIntelligenceExp;
-  gainStats = personMethods.gainStats;
-  regenerateHp = personMethods.regenerateHp;
-  updateSkillLevels = personMethods.updateSkillLevels;
-  hasAugmentation = personMethods.hasAugmentation;
+  gainHackingExp(exp: number): void {
```

Comment:
Although this is a potentially worthy refactor, let's save it for a different PR.